core/queryresultio/sparqlxml/src/main/java/org/eclipse/rdf4j/query/resultio/sparqlxml/SPARQLResultsSAXParser.java

```diff
@@ -132,12 +134,21 @@ public void startTag(String tagName, Map<String, String> atts, String text) thro
 			if (xmlLang != null) {
 				currentValue = valueFactory.createLiteral(text, xmlLang);
 			} else if (datatype != null) {
+				IRI datatypeIri;
 				try {
-					currentValue = valueFactory.createLiteral(text, valueFactory.createIRI(datatype));
+					datatypeIri = valueFactory.createIRI(datatype);
 				} catch (IllegalArgumentException e) {
 					// Illegal datatype URI
 					throw new SAXException(e.getMessage(), e);
 				}
+
+				// For broken SPARQL endpoints which return LANGSTRING without a language, fall back
+				// to using STRING as the datatype
+				if (RDF.LANGSTRING.equals(datatypeIri) && xmlLang == null) {
+					datatypeIri = XSD.STRING;
```

Comment:
We should perhaps log a warning here.

Reply:
Added a log message. Since this may happen often and there's not much one can do about it, I only log at debug level. Not sure how to properly contextualize the message though - maybe you could extend the message so that the user has some reasonable idea as to in relation to which result triple this is happening. Since a triple may have an arbitrary large value (e.g. for long description) some cutting down of the information might be called for. Probably you have best practices for this from other places in the code - but I am not familiar with them. Would be great if you could fix this up accordingly as required.

Reply:
Thanks. I'm actually fine with this as a log message (minus the small typo), though we could stick a fixed-length substring of the lexical value in there. I'll take a closer look.